### ISSUE TITLE
get.user_email() no longer works?

### ISSUE THREAD

**Issue by AvnerCohen** [KartikTalwar/gmail.js#734]
I can't seem to get user_email or manager_email to work.

Any chance there was a UI change that caused it to break?

**Comment by josteink** [comment]
I'm not able to reproduce this. Anyone else?

**Comment by AvnerCohen** [comment]
Thanks @josteink For checking that!
Here is what I am seeing, when stopping inside a `gmail.observe.on("recipient_change", ...` break point:
<img width="835" alt="image" src="https://user-images.githubusercontent.com/1297254/197190927-2b9a4b29-df53-4c4d-84e3-b36b6cd0fc2f.png">

**Comment by josteink** [comment]
Looking at the code, it seems all this is supposed to be extracted from `gmail.tracker.globals`.

How does that look on your end?

**Comment by AvnerCohen** [comment]
Seems empty .. 
<img width="760" alt="image" src="https://user-images.githubusercontent.com/1297254/197393852-fa4406d3-b223-4aa9-8d28-7cb5d361276e.png">

**Comment by josteink** [comment]
If so, that might make some other things stop working too…

If anyone can find an appropriate css-selector to fall back to, I guess we could add that as a fallback?

**Comment by AvnerCohen** [comment]
UI changed, using the following ugly-as-hell as a workaround:
```
  let mainObj = Array.from(document.querySelectorAll('div')).find(el => el.textContent === 'Google Account');
  var email = mainObj ? mainObj.parentNode.childNodes[2].innerText : 'N/A';
```

I am really surprised this is only happening to me...

**Comment by josteink** [comment]
If that works for you... How about making it into a PR? :smile:

**Comment by josteink** [comment]
If that works for you... How about making it into a PR? :smile: 

To be clear: I've tested it in the dev-console on my machine and it seems to work.

**Comment by AvnerCohen** [comment]
Yeah, but I think it's assume english as the gmail language, for example. Not something that is really scalable..
I am sure there are better ways to do that.
But I 1st love to know if this is even happening to other people..

**Comment by MattIrineu** [comment]
gmail.get.user_email() only returns undefined, it doesn't seem to work anymore, anyone with a workaround for it?

Thank you!

**Comment by josteink** [comment]
A short status update, based on a new duplicate bug-report.

Using the boilerplate and latest gmail.js, I've tested on three different google accounts, using Google Chrome, on Linux:

* gmail.com
* a grandfathered free google apps account
* and a "regular" paid google work account

On neither of those accounts can I reproduce this issue.

If this is an issue to you, you will probably have to do the debugging yourself, and try to reverse engineer another way to obtain the email and create a PR for it.

There's not really much I can do on my end.

**Comment by AvnerCohen** [comment]
Thanks @josteink yeah, it's been a while :) but I eventually got to try and find out what might be going on in here..
I got stuck pretty quickly I afraid..

I see the get method:
https://github.com/KartikTalwar/gmail.js/blob/master/src/gmail.js#L121

But I can't seem to find out where does this global ([10]) is ever being set...

Any suggested pointers ?

**Comment by josteink** [comment]
Globals are things set in the global window-namespace by Gmail itself. 

We dump it, investigate it, reverse-engineer it, and put that into our code. 

We never _set_ globals ourself.

**Comment by josteink** [comment]
Still not able to reproduce this issue. Closing for now.

Please re-open if you still have this issue, and know how to reproduce :)

**Comment by josteink** [comment]
Confirming I can reproduce this issue now.

**Comment by josteink** [comment]
Fix published in 1.1.13.
